Partest recognizes using

som-snytt opened the pull request:
Require line comment for current style pragma.

Other scala-cli style pragmas could be supported, but only option is recognized.

This simplifies posting code from reports, where dependencies are not involved.

som-snytt committed:
Partest recognizes using

Partest accepts options directive

lrytz commented on src/partest/scala/tools/partest/nest/Runner.scala:
```diff
@@ -480,11 +480,19 @@ class Runner(val testInfo: TestInfo, val suiteRunner: AbstractRunner) {
+          case namePattern(rest) => if (name == ToolName.filter) List(rest) else tokenize(rest)
+          case _ if name == ToolName.scalac =>
+            line match {
+              case optionsPattern(plural, rest) => if (plural == null) List(rest) else rest.split(',').toList.map(_.trim)
```
maybe for trailing whitespace?

```suggestion
              case optionsPattern(plural, rest) => if (plural == null) List(rest.trim) else rest.split(',').toList.map(_.trim)
```

som-snytt replied:
Now that you mention it, one reason to prefer `// scalac: args` is that it uses `tokenize`. `//> using options x, y, z` is actually unfortunate. `//> using options -Wconf:cat=deprecation:s,cat=feature:e` what does it do?

I will trim for now and explore the complications later.

I was glad to use one standard for `tokenize` everywhere, but like the xkcd, let's standardize on scala-cli and now we have two standards.

som-snytt replied:
I guess they support double-quotes. Maybe sufficient to tokenize then strip trailing comma, also simple comma.
```
tokenize("""-Xlint, "-Wconf:cat=deprecation:s,cat=feature:e", -feature , -deprecation""")
```

som-snytt replied:
Not trimming was cut/paste from `filter`, which probably should trim and not take the raw line. The filter would inadvertently not match with spurious spaces.

som-snytt commented:
Rebased and squashed, with improved handling of options. To avoid adding LOC, use tokenize but strip commas. Probably the limitation is that args can't end in comma.

The null test on optional group extraction is too bad. By coincidence, `regsafe` was just mentioned on dotty because it broke in the CB. Recall that `regextractor` would supply `Option` a decade ago. It's too bad that `regsafe` says that no one care about nulls, and also too bad that I'm checking for null in 2023.

som-snytt commented:
Or follow up with https://github.com/VirtusLab/using_directives thanks @SethTisue for the tip.
